### DISCUSSION TITLE
Storing a Firebase service-account.json file as file or secret?

### DISCUSSION THREAD

**Discussion by StefMa** [vercel/vercel#6503] [Help] [answered]
Hi there 👋 

I just ask myself "if its okay" to store the `serviceAccount.json` file next to the source code, or better put it into a [(secret) environemnt variable](https://vercel.com/docs/cli#commands/secrets)?

I found [this issue](https://github.com/vercel/vercel/issues/749) which is kind of similar what I ask here.
They discuss about how to store the google-json files in a secret.
But no one questions *if this is the best way* 😁 

Currently I have something like this:
```
- api
-- function.go
-- serviceAccount.json
```
My `vercel.json` file has something like this:
```json
{
  "version": 2,
  "functions": {
    "api/function.go": {
      "includeFiles": "serviceAccount.json"
    }
}
```

This works.
But not sure if this is "save" to do so.

Would be great if someone can answer this 👍

**Comment by leerob** [comment, marked as answer]
Yes, you should use Environment Variables!

Also, [Environment Variables](/docs/environment-variables) are now [encrypted by default](/changelog/environments-variables-per-git-branch), so it's no longer necessary to use `vercel secrets`. This command will eventually be deprecated 👍
